lib/chef/provider/zypper_repository.rb

```diff
@@ -0,0 +1,79 @@
+#
+# Author:: Tim Smith (<tsmith@chef.io>)
+# Copyright:: Copyright (c) 2017, Chef Software Inc.
+# License:: Apache License, Version 2.0
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+#
+
+require "chef/resource"
+require "chef/dsl/declare_resource"
+require "chef/mixin/which"
+require "chef/provider/noop"
+
+class Chef
+  class Provider
+    class ZypperRepository < Chef::Provider
+
+      extend Chef::Mixin::Which
+
+      provides :zypper_repository do
+        which "zypper"
```

Comment:
We really need to address this pattern, we do no caching AFAIK so this will run every time if there is more than one resource. Fine for now, but it's going to add up :(